pkg/sql/pg_extension.go

```diff
@@ -76,6 +72,17 @@ func postgisColumnsTablePopulator(
 						} else {
 							datumNDims = tree.DNull
 						}
+					default:
+						zm := m.ShapeType & geopb.ZShapeTypeFlag & geopb.MShapeTypeFlag
+						switch zm {
+						case geopb.ZShapeTypeFlag | geopb.MShapeTypeFlag:
+							datumNDims = tree.NewDInt(4)
+						case geopb.ZShapeTypeFlag, geopb.MShapeTypeFlag:
```

Comment:
the first case has a `|` and the second case has a `,` -- is that intentional?

Reply:
oh wait no, this is right. if Z&M are both set, i.e. `|`, it's 4, otherwise 3, etc.

Reply:
sneaky